crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -23,6 +26,7 @@ use serde::Serialize;
 #[derive(Default, TypeUuid)]
 #[uuid = "749479b1-fb8c-4ff8-a775-623aa76014f5"]
 pub struct DynamicScene {
+    pub resources: Vec<Box<dyn Reflect>>,
```

Comment:
Question: Have you considered using a `HashMap<TypeId, Box<dyn Reflect>>` over a `Vec`? This would prevent awkward situations where you have duplicate resources in `DynamicScene`.

Reply:
I'm not sure if this would be required for this PR since we have `Vec<DynamicEntity>` (which itself contains components as a `Vec`). Maybe this can be implemented in a future PR?

However, we probably should enforce uniqueness in `DynamicSceneBuilder` the same way we do for entities. I can see resources accidentally getting extracted multiple times.

Reply:
This is non-blocking, I approved the PR, just asking if it would be preferable.

Reply:
Gotcha. Sorry, glanced over the checkmark 😅

Reply:
I did not consider that, but as you explained, it definitely makes sense to avoid duplicate resources. As @MrGVSV also said, the same concepts applies to entity components as well, so for this reason I believe it should be done but in a separate PR.

Reply:
@Carbonhell Yeah I think the changes to `DynamicScene` can wait. However, I would suggest we make the proper changes to `DynamicSceneBuilder`.

This should either:
1. Disallow resources from being extracted more than once
2. Have future extractions overwrite existing resources
3. Clear and replace the entire collection of resource on each extraction

For simplicity and to be more aligned with components, I think we should go with Option 1 for now.

Reply:
Sorry for the extremely long delay and thank you for the feedback. As you said, I went with option 1 by changing the resources collection type from Vec to BTreeMap, so now they're handled in the same way of entities. The only small difference is that I used ComponentIds as keys, which from my understanding should be okay.

Reply:
Yeah I think that should be fine since we're not switching up the worlds during extraction.